actors/miner/tests/change_beneficiary_test.rs

```diff
@@ -0,0 +1,411 @@
+    let pending_beneficiary_term = info.pending_beneficiary_term.unwrap();
+    assert_eq!(beneficiary_change.expiration, pending_beneficiary_term.new_expiration);
+    assert_eq!(beneficiary_change.quota, pending_beneficiary_term.new_quota);
+    assert_eq!(first_beneficiary_id, pending_beneficiary_term.new_beneficiary);
+
+    //confirm proposal
+    h.change_beneficiary(
+        &mut rt,
+        first_beneficiary_id,
+        beneficiary_change.clone(),
+        Some(first_beneficiary_id),
+    )
+    .unwrap();
+
+    info = h.get_info(&rt);
+    assert_eq!(None, info.pending_beneficiary_term);
+    assert_eq!(first_beneficiary_id, info.beneficiary);
+    assert_eq!(beneficiary_change.quota, info.beneficiary_term.quota);
+    assert_eq!(beneficiary_change.expiration, info.beneficiary_term.expiration);
+
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_change_from_not_owner_beneficiary_to_another_address_three_message() {
+    let (mut h, mut rt) = setup();
+    let first_beneficiary_id = Address::new_id(999);
+    let second_beneficiary_id = Address::new_id(1001);
+
+    let first_beneficiary_term =
+        BeneficiaryTerm::new(TokenAmount::from(100), TokenAmount::zero(), ChainEpoch::from(200));
+    h.propose_approve_initial_beneficiary(&mut rt, first_beneficiary_id, first_beneficiary_term)
+        .unwrap();
+
+    let second_beneficiary_change = BeneficiaryChange::new(
+        second_beneficiary_id,
+        TokenAmount::from(101),
+        ChainEpoch::from(201),
+    );
+    h.change_beneficiary(&mut rt, h.owner, second_beneficiary_change.clone(), None).unwrap();
+    let mut info = h.get_info(&rt);
+    assert_eq!(first_beneficiary_id, info.beneficiary);
+
+    let mut pending_beneficiary_term = info.pending_beneficiary_term.unwrap();
+    assert_eq!(second_beneficiary_change.expiration, pending_beneficiary_term.new_expiration);
+    assert_eq!(second_beneficiary_change.quota, pending_beneficiary_term.new_quota);
+    assert_eq!(second_beneficiary_id, pending_beneficiary_term.new_beneficiary);
+    assert!(!pending_beneficiary_term.approved_by_beneficiary);
+    assert!(!pending_beneficiary_term.approved_by_nominee);
+
+    h.change_beneficiary(&mut rt, first_beneficiary_id, second_beneficiary_change.clone(), None)
+        .unwrap();
+    info = h.get_info(&rt);
+    assert_eq!(first_beneficiary_id, info.beneficiary);
+
+    pending_beneficiary_term = info.pending_beneficiary_term.unwrap();
+    assert_eq!(second_beneficiary_change.expiration, pending_beneficiary_term.new_expiration);
+    assert_eq!(second_beneficiary_change.quota, pending_beneficiary_term.new_quota);
+    assert_eq!(second_beneficiary_id, pending_beneficiary_term.new_beneficiary);
+    assert!(pending_beneficiary_term.approved_by_beneficiary);
+    assert!(!pending_beneficiary_term.approved_by_nominee);
+
+    h.change_beneficiary(&mut rt, second_beneficiary_id, second_beneficiary_change.clone(), None)
+        .unwrap();
```

Comment:
This test would be slightly better if we had the second beneficiary approve before the first beneficiary. The new beneficiary is always presumably in favour of the change, we want to show that they _also_ need the existing beneficiary to approve.